lambdas/validate-oauth-callback/src/main/java/uk/gov/di/ipv/core/validateoauthcallback/ValidateOAuthCallbackHandler.java

```diff
@@ -0,0 +1,203 @@
+import uk.gov.di.ipv.core.library.auditing.AuditEventUser;
+import uk.gov.di.ipv.core.library.config.ConfigurationVariable;
+import uk.gov.di.ipv.core.library.domain.ErrorResponse;
+import uk.gov.di.ipv.core.library.domain.JourneyResponse;
+import uk.gov.di.ipv.core.library.dto.ClientSessionDetailsDto;
+import uk.gov.di.ipv.core.library.dto.CredentialIssuerConfig;
+import uk.gov.di.ipv.core.library.dto.CredentialIssuerRequestDto;
+import uk.gov.di.ipv.core.library.dto.CredentialIssuerSessionDetailsDto;
+import uk.gov.di.ipv.core.library.dto.VisitedCredentialIssuerDetailsDto;
+import uk.gov.di.ipv.core.library.exceptions.HttpResponseExceptionWithErrorBody;
+import uk.gov.di.ipv.core.library.exceptions.SqsException;
+import uk.gov.di.ipv.core.library.helpers.ApiGatewayResponseGenerator;
+import uk.gov.di.ipv.core.library.helpers.LogHelper;
+import uk.gov.di.ipv.core.library.helpers.RequestHelper;
+import uk.gov.di.ipv.core.library.persistence.item.IpvSessionItem;
+import uk.gov.di.ipv.core.library.service.AuditService;
+import uk.gov.di.ipv.core.library.service.ConfigurationService;
+import uk.gov.di.ipv.core.library.service.IpvSessionService;
+
+public class ValidateOAuthCallbackHandler
+        implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final JourneyResponse JOURNEY_NEXT_RESPONSE =
+            new JourneyResponse("/journey/next");
+    private static final JourneyResponse JOURNEY_ERROR_RESPONSE =
+            new JourneyResponse("/journey/error");
+    private final ConfigurationService configurationService;
+    private final IpvSessionService ipvSessionService;
+    private final AuditService auditService;
+
+    public ValidateOAuthCallbackHandler(
+            ConfigurationService configurationService,
+            IpvSessionService ipvSessionService,
+            AuditService auditService) {
+        this.configurationService = configurationService;
+        this.ipvSessionService = ipvSessionService;
+        this.auditService = auditService;
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public ValidateOAuthCallbackHandler() {
+        this.configurationService = new ConfigurationService();
+        this.ipvSessionService = new IpvSessionService(configurationService);
+        this.auditService =
+                new AuditService(AuditService.getDefaultSqsClient(), configurationService);
+    }
+
+    @Override
+    @Tracing
+    @Logging(clearState = true)
+    public APIGatewayProxyResponseEvent handleRequest(
+            APIGatewayProxyRequestEvent input, Context context) {
+        LogHelper.attachComponentIdToLogs();
+
+        IpvSessionItem ipvSessionItem = null;
+        String credentialIssuerId = null;
+
+        try {
+            String ipvSessionId = RequestHelper.getIpvSessionId(input);
+
+            CredentialIssuerRequestDto request =
```

Comment:
If the intention is for this lambda to be called when the user is redirected with an Oauth error as well then I'm not sure if this part will work. Plus we are missing some of the useful features of the criError lambda. We would log what the oauth error was and set the visited credential issuer list etc.

Is this kind of thing going to be handled in another PR or needed within this one?

Reply:
Good point, for now we are still using the cri-error handler (called from the frontend).
Need to clarify whether we should be handling errors in here instead, if so will raise a separate subtask

Reply:
Yes, the idea is for this lambda to be the only one that is called when the user gets redirected to the callback url. So the frontend should not need to make any decisions as to whether the callback was an error or not.